front/lib/api/data_sources.ts

```diff
@@ -186,7 +187,9 @@ export async function deleteDataSource(
     }
   }
 
-  await dataSource.delete(auth);
+  await frontSequelize.transaction(async (t) => {
```

Comment:
It's okay to not wrap with a transaction here, as we can retry as long as the data source has not been destroyed.

Reply:
it may be OK but it's definitely worse not to wrap. If deleting the the data source fails, we're happy that we still have the views and the agent data source config

Reply:
I'm not entirely convinced because this code path is also used by our deletion process. This could result in the creation of some very long-lasting transactions, where recovery is not desirable.

Reply:
Why would it be long lasting ?

Reply:
Its only DB calls in there